### PR TITLE
Various minor refactorings

### PR DESCRIPTION
**Pull request by jahudzik**
Some minor code refactorings in order to make it more compact and readable:
- taking advantage of Java 7 features (diamond operator & multiple exceptions handling)
- extracted some code duplications
- fixed a typo

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -12,8 +12,8 @@ apply plugin: 'org.jetbrains.intellij'
 apply plugin: 'java'
 
 compileJava {
-    sourceCompatibility = 1.6
-    targetCompatibility = 1.6
+    sourceCompatibility = 1.7
+    targetCompatibility = 1.7
```
Comment (konifar) [inline build.gradle]: Java version should be 1.6 for Android Studio 😢

Comment (jahudzik) [reply build.gradle]: Could you tell why? Is it limitation from Android Studio?

I don't have much experience in Android Studio plugins development, but I managed to build your plugin with Java 1.7 (with some warnings) and run it successfully on Android Studio. Even creating a new IntelliJ Platform Plugin project on Java 1.8 worked fine for me.

Comment (konifar) [reply build.gradle]: Yes, it's limitetion of Android Studio.

As you said, Java 1.7 and 1.8 works well on IntelliJ. And Java 1.7 works on Android Studio in the case of debug. But when I create plugin jar file and load it to Android Studio, it doesn't work...

**Comment by konifar** [comment]
Thanks!!! Looks good to me!

**Comment by konifar** [comment]
And sorry to be late 🙇

**Pull request merged**